tests/check_framework/test_settings_deprecation.py

```diff
@@ -0,0 +1,23 @@
+from django.core import checks
+from django.core.checks.settings_deprecation import check_deprecated_settings
+from django.test import SimpleTestCase, override_settings
+
+
+class SettingsDeprecationCheckTests(SimpleTestCase):
+    @override_settings(TRANSACTIONS_MANAGED=True)
+    def test_deprecated_settings_variables(self):
+        warning_list = check_deprecated_settings()
```

Comment:
From looking at other tests of the check_framework, I think you should be using `checks.run_checks` here. This will fail if you use it now because your check is not registered.